### DISCUSSION TITLE
Backend responsive for openmage

### DISCUSSION THREAD

**Discussion by matteotestoni** [OpenMage/magento-lts#2125] [Ideas]
### Description (*)
I think for future openmage must adopt a css framework like bootstrap for a new backend theme and frontend responsive design.
For backend i see: http://demo4.cmsmart.net/admintheme/index.php/admin/
Maybe is possibile starting from this idea 
Maybe bootstrap 5

### Expected behavior (*)

backend and frontend modern responsive design

### Benefits

modernize openmage
### Additional information

i think best feautures:

- famous framework  (bootstrap, bulma, etc)
- seo friendly for fronend (not react)
- not reinvent the wheel like magento 2 so use a existent framework
- performance
- easy to migrate for openmage

**Comment by colinmollenhour** [comment]
Hi @matteotestoni, we simply don't have the resources to take on a complete rewrite of the CSS. We do have a new admin theme which fits nicely with the new OpenMage branding, although it needs a bit of work still.. Many complain about lack of contrast and there are a few other issues reported. I think it would be much more realistic to just make improvements to the new default theme, and if you would like to contribute towards that I'm sure everyone would appreciate it!

**Comment by loekvangool** [comment]
Perhaps it's more realistic to start replacing certain elements with more modern versions. I doubt that the mixed look and feel would really degrade the status quo as the backend is a hodgepodge after 10+ years of development.

If this is the way forward, we basically need to agree on a direction (like putting it in the "new admin theme"), agree on a library set (like Bootstrap 5), add the libraries and wait for PRs.

I myself have many wishes for the admin frontend, but they are all small/incremental and thus I would also not advocate for a complete rewrite.

**Comment by matteotestoni** [comment]
Hi  @loekvangool. For me http://demo4.cmsmart.net/admintheme/index.php/admin/ is a good starting point.
But for me the best is to discuss for a direction like bootstrap5 for backend and frontend for example.

**Comment by loekvangool** [comment]
That's a beautiful example @matteotestoni

**Comment by Flyingmana** [comment]
As making the theme responsive involves bigger changes, which possibly can conflict with existing custom theme people are using, I would like to have this as a new separate Theme. Especially if it involves usage of a new framework or similar.
It can still use the existing theme as base.

New themes can be started at https://github.com/OpenMage/OpenMage-Themes
having it separate allows us to apply changes more freely and less bureaucratic till they are ready for use.
You could see this as an incubator.
Maybe you can already kickstart this, by creating according themes there with an additional Goals.md file for each theme you would like to see.

**Comment by matteotestoni** [comment]
Hi @Flyingmana. Is possible discuss css framework with community? For example bootstrap?

**Comment by addison74** [comment]
My personal opinion is that OM Backend should have remained in the Magento version without the new theme being integrated and activated, which is not tested enough and can create issues. Whoever wanted to use it and improve it could do so in a separate project. From my experience when I switched to OM all content creators asked me to bring back the Magento theme. All these people work on desktops so a responsive feature goes unnoticed.

I would be satisfied if we managed to keep the interest for this project for a long time from now until people decide whether to opt for Magento 2 or something else. I know many situations of regret after migrating to Magento 2, I even have requests to return to M1. In addition, the attitude of some who are in the Magento 2 team and who are contemptuous of opinions and ideas, made me not embrace this platform. However, Varnish and ElasticSearch are required resources to reach somewhat acceptable performance.

Making a theme for OM Backend from scratch is a difficult task that consumes time and I don't think there is any interest in such a thing at the moment. Only one person I know is involved in the management of the current topic and receives little feedback. There are paid themes that are not bad at all. Bootstrap is a popular and powerful framework, I am more recently attracted to Tailwind where you don't touch the CSS file so often. I remain of the opinion that we should focus on improving the code and the performance, to bring basic lack features, to keep it compatible with the newest PHP and MySQL versions and to make it secure. 

Following the activity of the last year, I have identified around 20 people who are active and on whom we can still rely. I doubt that one of them would want to dedicate himself to creating a new theme for OM Backend.

**Comment by addison74** [comment]
Not only a CSS framework needs OM but also a detachment of the Prototype library as it happened with Magento 2 where JQuery was chosen. It remains a dream and I remain of the opinion that we do not have to change something that works

**Comment by Danieleeffe1** [comment]
there was a software house that had created an adaptive backand, now the site no longer exists.

mgtextensions.com

I bought it and use it, it's not perfect but you can work on it by taking it as a base.

**Comment by aterjung** [comment]
I want to add some info why the new om backend theme is build the way it is, and not with a framework like bootstrap.
The goal i had in mind was to reduce changes to the template files where ever possible. You have to have the extensions in mind. Many Extensions bring there own backend templates or at least customise the backend by replacing templates with modified copys of the original. They all depend on the css structure of the magento standard theme - there was no reason not to do so cause it was not replaceable by default and was not changed over 10 years.
Looking at themes like the bootstrap one is great, but it will not longer look great if you add some extensions. Even things like Paypal Plus, or at least the AOE Scheduler will break with this theme.
Its sad, but i can`t imagine how we should get enough manpower to maintain exceptions to make all and every extension work again.
Thats the reason i did not touch the underlying template files (except some) and tried to do everything with css.
I would highly appreciate any help improving the theme. It is far from perfect, but it helps openmage to be seen more independent.
Probably we should start an new issue to collect ideas to improve the admin theme at first.

**Comment by Danieleeffe1** [comment]
> Voglio aggiungere alcune informazioni sul motivo per cui il nuovo tema di backend om è costruito così com'è e non con un framework come bootstrap.
> L'obiettivo che avevo in mente era ridurre le modifiche ai file modello ove possibile. Devi avere in mente le estensioni. Molte estensioni portano i propri modelli di backend o almeno personalizzano il backend sostituendo i modelli con copie modificate dell'originale. Dipendono tutti dalla struttura css del tema standard di Magento - non c'era motivo per non farlo perché non era sostituibile per impostazione predefinita e non è stato modificato in 10 anni.
> Guardare temi come quello bootstrap è fantastico, ma non sarà più bello se aggiungi alcune estensioni. Anche cose come Paypal Plus, o almeno AOE Scheduler, romperanno con questo tema.
> È triste, ma non riesco a immaginare come dovremmo ottenere abbastanza manodopera per mantenere le eccezioni per far funzionare di nuovo tutte le estensioni.
> Questo è il motivo per cui non ho toccato i file modello sottostanti (tranne alcuni) e ho provato a fare tutto con css.
> Apprezzerei molto qualsiasi aiuto per migliorare il tema. È tutt'altro che perfetto, ma aiuta openmage a essere visto più indipendente.
> Probabilmente dovremmo iniziare un nuovo numero per raccogliere idee per migliorare il tema dell'amministratore all'inizio.

Hello

As I wrote in my previous post I bought an adaptive theme for the backend

I created a demo user to give you a chance to see it.

I hope it will be useful and inspiring for you.

It works on magento 1.9.4.4 but I have installed it without problems also on OpenMage 20.0.13.

Please don't mess with my Magento installation

https://monsterbox.it/admin

user:
demo

passwd:
DemoAdattiva123

**Comment by aterjung** [comment]
Thank you for sharing, did you forget to add the url, or did I just miss something from a previous post?

**Comment by Danieleeffe1** [comment]
> Grazie per la condivisione, ti sei dimenticato di aggiungere l'URL o mi sono perso qualcosa di un post precedente?


yes I forgot it and I updated the post

**Comment by Danieleeffe1** [comment]
Attention!!

 on 15/10/2021 the server with the magento installation where you can see the responsive administrative section will be closed.

**Comment by sreichel** [comment]
@Danieleeffe1 if that code is still available, please share.

Closed for now.
